notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java

```diff
@@ -907,6 +901,17 @@ public void removeDraftOfNote(Page page, String username) throws WikiException {
    */
   @Override
   public void removeDraftOfNote(Page page) throws WikiException {
+    List<DraftPage> draftPages = dataStorage.findDraftPagesByTargetPage(page);
+    for (DraftPage draftPage : draftPages) {
+      if (draftPage != null) {
+        dataStorage.deleteAttachmentsOfDraftPage(draftPage);
+        try {
+          deleteNoteMetadataProperties(draftPage, draftPage.getLang(), NOTE_METADATA_DRAFT_PAGE_OBJECT_TYPE);
```

Comment:
I prefer that the properties deletion come after the draft deletion itself, so when there is a problem with the draft deletion the properties will not be deleted

Reply:
https://github.com/Meeds-io/notes/pull/1094#discussion_r1753509503